promql_tester/condition.py

```diff
@@ -0,0 +1,110 @@
+"""Condition object encapsulate condition attributes"""
+import uuid
+import yaml
+
+
+class MalformedConfigException(Exception):
+    """Exception thrown when malformed configuration file is handled"""
+
+
+class FailedCondition(Exception):
+    """Exception thrown when a condition fails"""
+
+
+class ConditionConfigException(Exception):
+    """Exception thrown when condition configuration is not valid"""
+
+
+class Condition:
+    """Condition class represent a condition that needs to be evaluated.
+    `expression` - promql expression to be evaluated
+    `for_seconds` - time period to evaluate the condition for
+    `interval_seconds` - the condition will be evaluated every interval_seconds seconds
+    `threshold` - threshold for the expression's result to be in for the condition to be valid
+    `threshold.upper` - upper threshold
+    `threshold.lower` - lower threshold
+    """
+    REQUIRED_FIELDS = ["expression", "for_seconds", "interval_seconds"]
+    DEFAULT_THRESHOLD = {"upper": 0, "lower": 0}
+    _expression: str
+    _for_seconds: str
+    _interval_seconds: str
+    _threshold: dict
+
+    def __init__(self, raw_config: dict):
+        """
+        raw_config: dictionary representing parsed yaml. Expected to have
+        `expression`, `for_seconds`, `interval_seconds` keys.
+        Optionally can have `threshold` key, which is expected to contain a dictionary
+        with `upper` and `lower` keys.
+        `expression` is a promql expression
+        `for_seconds` are seconds that the condition needs to be evaluated for
+        `interval_seconds` represent the interval of which the condition will be evaluated
+        `threshold` is the threshold for which the value is expected to be in. Defaults to
+        `upper` = 0 and `lower` = 0, meaning that if the result is 0 the condition is successful.
+        """
+        self._raise_if_error(raw_config)
+        self._expression = raw_config["expression"]
+        self._for_seconds = int(raw_config["for_seconds"])
+        self._interval_seconds = int(raw_config["interval_seconds"])
+        self._uuid = uuid.uuid4()
+        self._threshold = self._get_threshold(raw_config)
+
+    def get_id(self) -> str:
+        """Returns autogenerated condition ID"""
+        return self._uuid
+
+    def get_interval_seconds(self):
```

Comment:
I would suggest to have properties instead of getters/setters
```suggestion
    @property
    def interval_seconds(self):
```

Reply:
I can't seem to grasp what are the advantages of this approach, is this just more pythonic or is there more to it?

Reply:
For Python developers if it looks like an object property they prefer getting it as such (instead of Java let's say that prefers declaring it as part of the object's behavior)

Reply:
yes, it's more pythonic. From what I know, in python the attributes are public by default, we access to them via `my_class.my_attr`. The `@property` decorator allows us to encapsulate the attributes inside the class while keeping the same access logic.